src/app/services/technical-record/technical-record.service.ts

```diff
@@ -416,6 +422,57 @@ export class TechnicalRecordService {
     };
   }
 
+  validateForBatch(): AsyncValidatorFn {
```

Comment:
I might be worth seperating this method into two private methods, one in case `trailerId && vin`, one in case `!trailerId && vin`. It's just nicer than a 30-40 line method